arroyo/backends/kafka/commit.py

```diff
@@ -41,13 +40,9 @@ def decode_legacy(self, value: KafkaPayload) -> Commit:
             raise TypeError("payload value must be a bytes object")
 
         headers = {k: v for (k, v) in value.headers}
-        try:
-            orig_message_ts: Optional[datetime] = datetime.strptime(
-                headers["orig_message_ts"].decode("utf-8"), DATETIME_FORMAT
-            )
-        except KeyError:
-            orig_message_ts = None
-
+        orig_message_ts = datetime.strptime(
+            headers["orig_message_ts"].decode("utf-8"), DATETIME_FORMAT
+        )
```

Comment:
This code (decode_legacy) will eventually all be deleted anyway - I'm waiting for the next self hosted release to make the migration smoother